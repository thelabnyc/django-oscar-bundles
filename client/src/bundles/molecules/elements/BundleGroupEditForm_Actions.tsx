import React from "react";

interface IProps {
    onCancel: (e: React.MouseEvent<HTMLElement>) => void;
}

export class BundleGroupEditFormActions extends React.PureComponent<IProps> {
    render() {
        return (
            <div className="col-sm-12 bundle-group-edit__section">
                <div className="form-group">
                    <a
                        className="btn btn-secondary"
                        onClick={this.props.onCancel}
                    >
                        {gettext("Cancel")}
                    </a>{" "}
                    <button className="btn btn-primary" type="submit">
                        {gettext("Save")}
                    </button>
                </div>
            </div>
        );
    }
}
