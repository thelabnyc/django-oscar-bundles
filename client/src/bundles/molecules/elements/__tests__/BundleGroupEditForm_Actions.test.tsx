import React from 'react';
import {render, screen} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import {BundleGroupEditFormActions} from '../BundleGroupEditForm_Actions';


describe('#bundles/molecules/elements/BundleGroupEditFormActions', () => {

    it('renders', () => {
        const wrapper = render(
            <BundleGroupEditFormActions
                onCancel={() => null}
            />
        );
        expect(wrapper.asFragment()).toMatchSnapshot();
    });


    it('triggers close callback', () => {
        const onCancel = jest.fn();
        render(
            <BundleGroupEditFormActions
                onCancel={onCancel}
            />
        );

        expect(onCancel).toHaveBeenCalledTimes(0);

        userEvent.click(screen.getByText('Cancel'));

        expect(onCancel).toHaveBeenCalledTimes(1);
    });

});
