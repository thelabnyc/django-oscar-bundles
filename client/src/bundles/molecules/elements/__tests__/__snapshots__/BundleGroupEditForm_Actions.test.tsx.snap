// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#bundles/molecules/elements/BundleGroupEditFormActions renders 1`] = `
<DocumentFragment>
  <div
    class="col-sm-12 bundle-group-edit__section"
  >
    <div
      class="form-group"
    >
      <a
        class="btn btn-secondary"
      >
        Cancel
      </a>
       
      <button
        class="btn btn-primary"
        type="submit"
      >
        Save
      </button>
    </div>
  </div>
</DocumentFragment>
`;
